R/servers.R

```diff
@@ -105,7 +102,7 @@ cloudServerInfo <- function(name = "posit.cloud") {
 #' @export
 discoverServers <- function(quiet = FALSE) {
   # TODO: Better discovery mechanism?
-  discovered <- getOption("rsconnect.local_servers", "http://localhost:3939/__api__")
+  discovered <- getOption("rsconnect.local_servers", NULL)
```

Comment:
Yet another feature I had never encountered. I do not know what "auto-discovery" means or how it might be configured by administrators. The old default _probably_ uses that value in case RSP and RSC are installed onto the same server, but I don't have enough context to be sure.

First landed: https://github.com/rstudio/rsconnect/commit/83642fca83e7bd8f8f0ee45a521a0e1ab8390b16

I can find no references outside the `rsconnect` repository and cannot find any documentation for the `rsconnect.local_servers` option.

My vote is to deprecate this function unless we can find additional context.

Reply:
It looks like it was added to facilitate testing, and was never documented, so I agree that deprecating seems safe. I'll do that in a separate PR to avoid making this one more complicated.

Reply:
Now in #737